client/src/endpoint/ledger.rs

```diff
@@ -170,38 +203,185 @@ impl IntoRequest for Details {
 }
 
 #[cfg(test)]
-mod all_ledgers_tests {
+mod ledger_details_tests {
+    use super::*;
+
+    #[test]
+    fn it_can_make_a_ledger_details_uri() {
+        let details = Details::new(12345);
+        let request = details
+            .into_request("https://horizon-testnet.stellar.org")
+            .unwrap();
+        assert_eq!(request.uri().host().unwrap(), "horizon-testnet.stellar.org");
+        assert_eq!(request.uri().path(), "/ledgers/12345");
+    }
+}
+
+/// Represents the payments for ledger endpoint on the stellar horizon server.
+/// The endpoint will return all the payment for a single ledger in the chain.
+///
+/// <https://www.stellar.org/developers/horizon/reference/endpoints/payments-for-ledger.html>
+///
+/// ## Example
+/// ```
+/// use stellar_client::sync::Client;
+/// use stellar_client::endpoint::{ledger, payment, transaction};
```

Comment:
Btw, I decided to make this one test much more detailed in lieu of the other examples. If ya'll like this then I think this is a pattern to follow?

![image](https://user-images.githubusercontent.com/2043529/38263546-35b83974-3725-11e8-809d-5e3052f3a19b.png)

I think it's nice because it demonstrates the relationship between payments, ledgers, and transactions.

Reply:
Wow.  Nice.  It is also the only real way to ensure that a request would yield a payment.  I am 100% in favor of this moving forward.